tests/php/test-amp-image-dimension-extract-download.php

```diff
@@ -12,31 +12,90 @@
  */
 class AMP_Image_Dimension_Extract_Download_Test extends WP_UnitTestCase {
 
+	private static $pid;
+
 	const AMP_IMG_DIMENSION_TEST_INVALID_FILE = __DIR__ . '/assets/not-exists.png';
 
 	public static function setUpBeforeClass() {
 		parent::setUpBeforeClass();
+		self::start_server();
+	}
+
+	public static function tearDownAfterClass() {
+		parent::tearDownAfterClass();
+		self::stop_server();
+	}
+
+	/**
+	 * Start PHP built-in server to host images to test.
+	 */
+	private static function start_server() {
+		// Not ideal to use remote URLs; mocking would be better for performance, but FasterImage doesn't provide means to do this.
+		$url = wp_parse_url( self::get_server_url() );
 
-		/*
-		 * Start PHP built-in server to host images to test.
-		 * Not ideal to use remote URLs; mocking would be better for performance, but FasterImage doesn't provide means to do this.
-		 */
-		$host      = self::get_host();
-		$host_root = self::get_host_root();
+		if ( ! isset( $url['host'], $url['port'] ) ) {
+			throw new \Error( 'A host and port needs to be set to start the PHP server' );
```

Comment:
What is `Error`? Do you mean `Exception`?

```suggestion
			throw new \Exception( 'A host and port needs to be set to start the PHP server' );
```

Reply:
Yes :man_facepalming:.